packages/task/src/task.ts

```diff
@@ -37,11 +37,11 @@ export const initialState = Symbol();
 
 export type TaskStatus = (typeof TaskStatus)[keyof typeof TaskStatus];
 
-export type StatusRenderer<R> = {
-  initial?: () => unknown;
-  pending?: () => unknown;
-  complete?: (value: R) => unknown;
-  error?: (error: unknown) => unknown;
+export type StatusRenderer<R, D extends ReadonlyArray<unknown>> = {
```

Comment:
We call the args type `T` elsewhere, so I'd either stick with that or give all the type parameters better names like `Result` and `Args`.